### ISSUE TITLE
Can not change Color of LightComponent programmatically

### ISSUE THREAD

**Issue by wanderwurst** [SiliconStudio/paradox#296]
I thought the Color property of a LightComponent would allow me to change the Color at runtime?

Here is what I do:

I set up a fresh new project (official Beta 1.2.2, Win7 VS2015) and add a LightComponent to the scene (type doesnt matter, Point, Spot...I tried all of them). I set the Color, Size, Intensity etc. so that the light is clearly visible (e.g. a green spot on the ground).

In VS I add this script which I attach to the light:

```C#
using SiliconStudio.Paradox.Engine;

namespace LightTest
{
    public class ColorChanger : SyncScript
    {
        private LightComponent _light;

        public override void Start()
        {
            _light = this.Entity.Get<LightComponent>();
        }

        private int frameCount = 0;

        public override void Update()
        {
            if (frameCount++ == 60)
            {
                frameCount = 0;
                _light.Color = new SiliconStudio.Core.Mathematics.Color3(255, 0, 0);
                _light.Update(); // Just for demo purpose
            }
        }
    }
}
```

Each second, this script tries to set the color of the light to red (at least that is my intention).
In the debugger, i see that the Color is indeed 255,0,0 right after the assignment.
However in the next cycle (or right after calling the _light.Update()) the Color is reset to what I originally chose in ParadoxStudio.
The light on screen never changes color.

Am I just doing it wrong? If so, how can I change a LightComponent.Color at runtime?

**Comment by xoofx** [comment]
The field `LightComponent.Color` is actually used internally to store computed color and is calculated by the `LightProcessor` (which is running after your script, hence why your change doesn't make it).

We will remove these fields from public to avoid confusing usage.

In order to access the color, you go through the `LightComponent.Type` and then `LightColorRgb`:
```
((LightColorRgb)((IColorLight)_light.Type).Color).Color = Color.Red;
```
We should probably add a bit of extension methods to ease this kind of scenario though (Also the naming is a bit annoying currently, as we have `ILightColor` and `IColorLight`). The reason for this model is that we are planning to add also different color models like temperature color at some point.

**Comment by wanderwurst** [comment]
Thanks a lot for the quick feedback!
I dug the source meanwhile and found this as well, works a treat.

As for the planned changes: I would probably find it more intuitive to have the properties by the same name as I they appear in the ParadoxStudio, so maybe keeping the Color property and just adding getters/setters to the underlying structures would be a good idea? (and then instead rename the current Color to something like ComputedColor) 
Thus it would be coherent with the other properties like Intensity etc. (which makes life easier if you dont want to consult the source on things like that ;-)
But then again, that's just me, so thanks again for the quick help.

**Comment by xoofx** [comment]
On next release, their will be extensions methods `GetColor/SetColor` accessible on the `LightComponent` directly. Also `ILightColor` has been renamed to `IColorProvider` as well as `ColorRgbProvider` and moved outside the Light namespace.

**Comment by wanderwurst** [comment]
Thank you very much

**Comment by wanderwurst** [comment]
Hm, the changes (in 1.3.1) don't seem to work for me yet.
I have a LightComponent myLight which is a Point Light. If I call 
```C#
myLight.SetColor(new Color3(0.5f, 0.4f, 0.3f));
```
I get this
```C#
InvalidOperationException
{"The LightComponent doesn't contain a color light type IColorLight"}
```

However I can easily set it myself by using the new naming conventions like so:
```C#
((ColorRgbProvider)((IColorLight)(myLight.Type)).Color).Value = new (new Color3(0.5f, 0.4f, 0.3f);
```
so it's probably just something in the new Setter extension?

**Comment by wanderwurst** [comment]
I see it's been fixed in 1.3.3, thanks!
